packages/wrangler/src/deploy/index.ts

```diff
@@ -244,8 +248,29 @@ export const deployCommand = createCommand({
 		const projectRoot =
 			config.userConfigPath && path.dirname(config.userConfigPath);
 
-		const entry = await getEntry(args, config, "deploy");
+		if (!config.configPath) {
+			// Attempt to interactively handle `wrangler deploy <directory>`
+			if (args.script) {
+				try {
+					const stats = statSync(args.script);
+					if (stats.isDirectory()) {
+						args = await handleMaybeAssetsDeployment(args);
```

Comment:
Consider moving the dir check inside `handleMaybeAssetsDeployment` ?

Maybe comment on `handleMaybeAssetsDeployment` what the req are (i.e. directory)